### PR TITLE
new javascript asset handling

### PR DESCRIPTION
**Pull request by xithan**

### CODE REVIEW

**Comment by xithan** [comment]
@ethn Please double check the coffee files. I moved all files but I haven't changed anything. If you have recently changed something there, it could be that I reverted it accidentally when I merged with master.

File: card/lib/cardio/mod/loader/set_loader.rb [hunk 1]
```diff
@@ -10,7 +10,7 @@ class SetLoader < Loader
 
         def initialize args={}
           @patterns = args.delete(:patterns) || Card::Set::Pattern.loadable_codes
-          super
+          super load_strategy: args[:load_strategy], mod_dirs: args[:mod_dirs]
```
[the remark this answers is not in the document]
Comment (xithan) [reply card/lib/cardio/mod/loader/set_loader.rb:13]: No, that is to make it work with ruby 3 (https://www.ruby-lang.org/en/news/2019/12/12/separation-of-positional-and-keyword-arguments-in-ruby-3-0).
I didn't fix all of these issues and switched back to ruby 2.7 instead.

File: card/lib/card/mod.rb [hunk 2]
```diff
@@ -0,0 +1,65 @@
+class Card
```
[the remark this answers is not in the document]
Comment (xithan) [reply card/lib/card/mod.rb:1]: That is not the MODULE Cardio::Mod. Is is a new CLASS. But maybe we need a different name for less confusion

File: card/lib/card/model/save_helper.rb [hunk 3]
```diff
@@ -60,7 +60,7 @@ def delete_code_card name
 
         card = Card[name]
         card.update! codename: nil
-        card.delete!
+        card.delete! update_referers: true
```
[the remark this answers is not in the document]
Comment (xithan) [reply card/lib/card/model/save_helper.rb:63]: Deleting script cards didn't remove the references in *all+*script. But this didn't help.

File: card/mod/core/set/all/actify.rb [hunk 4]
```diff
@@ -0,0 +1,68 @@
+def act options={}, &block
```
[the remark this answers is not in the document]
Comment (xithan) [reply card/mod/core/set/all/actify.rb]: ok, I merged and removed this

File: decko/lib/decko/tasks/decko.rake [hunk 5]
```diff
@@ -110,6 +110,21 @@ decko_namespace = namespace :decko do
...
+  task mod_install: :environment do
+    Card.reset_script_machine
+    Card::Cache.reset_all
+    Cardio::Mod.dirs.mods.each do |mod|
+      mod.ensure_mod_card
+      Card::Cache.reset_all
+    end
+  end
+
+  desc "install mods"
```
[the remark this answers is not in the document]
Comment (xithan) [reply decko/lib/decko/tasks/decko.rake]: yes , removed

**Review by ethn — APPROVED** [review]

**Pull request merged**